cmd/eksctl/main.go

```diff
@@ -26,10 +27,15 @@ var rootCmd = &cobra.Command{
 
 func init() {
 
-	addCommands()
+	g := cmdutils.NewGrouping()
 
+	addCommands(g)
+
+	rootCmd.PersistentFlags().BoolP("help", "h", false, "help for this command")
 	rootCmd.PersistentFlags().IntVarP(&logger.Level, "verbose", "v", 3, "set log level, use 0 to silence, 4 for debugging and 5 for debugging with AWS debug logging")
 	rootCmd.PersistentFlags().BoolVarP(&logger.Color, "color", "C", true, "toggle colorized logs")
+
+	rootCmd.SetUsageFunc(g.Usage)
```

Comment:
We have to set this here, setting it at each subcommand's level doesn't actually work, although it's not clear why from the implementation.